src/filebrowser/listing.ts

```diff
@@ -882,9 +874,7 @@ class DirListing extends Widget {
     }
 
     let item = this._model.sortedItems[i];
-    this._model.open(item.name).catch(error =>
-        utils.showErrorMessage(this, 'File Open Error', error)
-    );
+    this._registry.open(item);
```

Comment:
If item is a directory, the registry open function short-circuits and returns. I think we should special-case directories here and navigate to them. As it is, double-clicking on a directory does nothing for me.